tool/tsh/common/app_azure.go

```diff
@@ -231,23 +232,17 @@ func (a *azureApp) startLocalALPNProxy(port string) error {
 		return trace.Wrap(err)
 	}
 
-	// backend expects the tokens to be signed with web session private key
-	ws, err := tc.GetAppSession(a.cf.Context, types.GetAppSessionRequest{SessionID: a.app.SessionID})
-	if err != nil {
-		return trace.Wrap(err)
-	}
-
-	wsPK, err := utils.ParsePrivateKey(ws.GetPriv())
-	if err != nil {
-		return trace.Wrap(err)
+	signer, ok := appCerts.PrivateKey.(crypto.Signer)
+	if !ok {
+		return trace.BadParameter("private key type %T does not implement crypto.Signer", appCerts.PrivateKey)
```

Comment:
Can we add some hints to the error message - what should the user do in such case?

Reply:
Realistically this should never fail, `appCerts.PrivateKey` should always cast to `crypto.Signer`.

```go
// "crypto" package
// 
// PrivateKey represents a private key using an unspecified algorithm.
//
// Although this type is an empty interface for backwards compatibility reasons,
// all private key types in the standard library implement the following interface
//
//	interface{
//	    Public() crypto.PublicKey
//	    Equal(x crypto.PrivateKey) bool
//	}
//
// as well as purpose-specific interfaces such as [Signer] and [Decrypter], which
// can be used for increased type safety within applications.
type PrivateKey any
```

We just do the check to ensure there isn't an unexpected panic.